### ISSUE TITLE
Use common github workflow for Jekyll

### ISSUE THREAD

**Issue by andrewpatto** [ga4gh/data-security#100]
We currently have duplicated the Jekyll publish workflow into one pipeline for drafts and one for main (with minor differences).

We could refactor the common workflow bits into a shared workflow that is callable from the two workflows.

**Comment by TomConner** [comment]
Yes and it would be nice to allow main to be updated without deleting everything -- it could preserve an allowlist of versions and branches and delete everything not on the list.
